### DISCUSSION TITLE
Angular application - Live reload doesn't work behind YARP proxy

### DISCUSSION THREAD

**Discussion by prabh-62** [dotnet/yarp#1954] [Q&A] [answered]
Hello, I created a YARP based reverse proxy to route multiple angular applications and .NET 6 web APIs. I encountered an issue during this process. If I don't use proxy, any edits I make to angular apps are automatically refreshed in browser (Live reload). If I use proxy, then websocket call fails(which is used by live-reload). 

I asked for help from angular team https://github.com/angular/angular-cli/issues/24373 and they concluded that with nodejs-based proxy, websocket calls succeed so it could be an issue with YARP configuration. I created a simple git repo to showcase the issue - https://github.com/prabh-62/angular-15-proxy-websocket-url-bug

Do you know if I my YARP configuration is incorrect?

**Comment by samsp-msft** [comment, marked as answer]
Looking at the picture, the ws socket is to ws://localhost/ng-cli-ws, but I don't see that in the route in the proxy configuration. The app doesn't appear to be using a relative path.

**Reply by prabh-62** [reply]
Thank you that was exactly the issue. I had to specify the `--serve-path` option when launching angular apps

I edited  `web-apps/package.json`
```json
    "start:command-center": "ng serve --project command-center --port 4201 --serve-path /command-center",
    "start:support-panel": "ng serve --project support-panel --port 4202 --serve-path /support-panel"
```

and removed `transforms` from YARP config.
